lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,31 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+  # Gives the existing lookup paths that are suffixes of the fully qualified path of `klass`.
+  # @param [Object] klass
+  # @return [Array<Symbol>]
+  def nested_lookup_paths(klass)
+    paths = klass.to_s.split('::').reverse.reduce([]) { |a, e| a + [[e] + Array(a.last)] }.reverse
+    paths.map! { |path| "#{path.join('::')}" }
+    paths.map! do |path|
```

Comment:
I'll comment the first line, which might be pretty cryptic. The rest is pretty much self-explanatory, no?